test/test_declare_v2_forked.py

```diff
@@ -86,3 +108,43 @@ def test_declare_v2_and_get_class_by_hash():
         expected_path=CONTRACT_1_CASM_PATH,
         feeder_gateway_url=APP_URL,
     )
+
+
+@devnet_in_background("--port", ORIGIN_PORT, *PREDEPLOY_ACCOUNT_CLI_ARGS)
+def test_declare_v2_if_already_declared_on_origin():
+    """Declare on origin; declare on fork; assert rejected"""
+
+    # declare on origin
+    contract_class, _, compiled_class_hash = load_cairo1_contract()
+
+    declaration_resp = send_declare_v2(
+        contract_class=contract_class,
+        compiled_class_hash=compiled_class_hash,
+        sender_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        sender_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+        gateway_url=ORIGIN_URL,
+    )
+    assert_declare_v2_accepted(declaration_resp, feeder_gateway_url=ORIGIN_URL)
+
+    # fork
+    FORKING_DEVNET.start(
+        "--port", FORK_PORT, "--fork-network", ORIGIN_URL, "--accounts", "0"
+    )
+
+    # if forking devnet with devnet, as it currently is in this test, the following assertion would
```

Comment:
do we still need that code?

Reply:
I left it so that anyone who ever considers adding it again gives up

Reply:
So even if someone will try to add it will fail and this code will not be added because is flaky in this scenario. And by keeping it we try to avoid this burned time? I think that it's really unlikely to happen.

I just think that keeping commented code in the code base is anti-pattern but it's your call - treat it as a suggestion.

Reply:
I can replace the commented code with a commented sentence of explanation.